packaging/release.py

```diff
@@ -1345,8 +1345,11 @@ def test_sdist() -> None:
                 sdist = stack.enter_context(tarfile.open(sdist_file))
             except FileNotFoundError:
                 raise ApplicationError(f"Missing sdist: {sdist_file.relative_to(CHECKOUT_DIR)}") from None
-
-            sdist.extractall(temp_dir)
+            if hasattr(tarfile, 'data_filter'):
+                # Remove this check when Python 3.11 is EOL
```

Comment:
Use a deprecation comment here (implemented by our own `deprecated` plugin for `pylint`):

```suggestion
                # deprecated: description='extractall fallback without filter' python_version='3.11'
```